### ISSUE TITLE
timeToRands produces constant sequence if seed is 0 => replace tidal's random mechanism with System.Random?

### ISSUE THREAD

**Issue by jwaldmann** [tidalcycles/Tidal#1067]
```
ghci> import Sound.Tidal.UI
ghci> timeToRands 0 10
[0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]
```
this function is used in `randrun`, which is used in `shuffle`, so we get (for query arc starting at 0)
```
ghci> shuffle 4 $ run 4
(0>¼)|0
(¼>½)|1
(½>¾)|2
(¾>1)|3
```
the reason is that the implementation iterates `xorwise` which maps 0 to 0 (as all xor/shift generators do).

perhaps `xorwise` can be replaced with something from System.Random. E.g., think of `mkStdGen` as `timeToIntSeed`, then  the first value already is "random" (non-zero).
```
ghci> import System.Random
ghci> uniformR (0,10) (mkStdGen 0)
(8,StdGen {unStdGen = SMGen 16294208416658607535 16294208416658607535})
```

**Comment by sss-create** [comment]
hm, what to do @yaxu. Would changing this break sequences people built and rely on or would this be a welcoming fix?

**Comment by yaxu** [comment]
This algorithm was chosen by @dktr0, after hitting performance issues with what we were using before (System.Random.MWC). https://github.com/tidalcycles/Tidal/pull/640

We can change the algorithm as a breaking change but it should be carefully profiled as it's a bit of a bottleneck. Perhaps there's a simpler tweak to avoid seeding with 0 though..

**Comment by dktr0** [comment]
If I am following this correctly, the concern is that when and only when multiple random values are requested at exactly time point 0, the results are all the same.

I believe the simple fix might just be to replace filter/replace 0 as a seed with another equally definite value (for example: the number 1)? I think that would have no performance implications, and would preserve 99.99% of any deterministic results people are "relying on".

(Of course, the results would still always, everywhere be a deterministic function of metre, and I think that's a very good thing, since it allows experientially similar - even "identical" - results across remotely networked ensembles.)

**Comment by yaxu** [comment]
Yes that seems an easy fix. Maybe not 1, as then I think events at consecutive cycle onsets would be the same. Any other number above 9, probably no-one would notice :)

(Agreed determinism is good.)

**Comment by dktr0** [comment]
Perhaps a really large number instead of 1, then, "to be sure", like millions of cycles into a timeline...

**Comment by sss-create** [comment]
closing with #1142
